File: backend/app/schemas/ai_usage.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AIUsagePublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    admin_id: int | None
    source: str
    model: str
    input_tokens: int
    output_tokens: int
    estimated_cost_usd: float
    outcome: str
    created_at: datetime


class AIUsageByAdmin(BaseModel):
    admin_id: int | None
    username: str
    estimated_cost_usd: float


class AIUsageDaily(BaseModel):
    date: str
    estimated_cost_usd: float


class AIUsageSummary(BaseModel):
    budget_usd: float
    budget_rm_display: float
    spent_usd: float
    spent_rm_display: float
    remaining_usd: float
    percent_used: float
    warning_threshold_percent: float
    warning: bool
    month_start: datetime
    by_admin: list[AIUsageByAdmin]
    daily: list[AIUsageDaily]

File: frontend/src/api/aiUsage.ts
import api from "./axios";

export interface AIUsageRecord {
  id: number;
  admin_id: number | null;
  source: string;
  model: string;
  input_tokens: number;
  output_tokens: number;
  estimated_cost_usd: number;
  outcome: "RESERVED" | "COMPLETED" | "FAILED";
  created_at: string;
}

export interface AIUsageSummary {
  budget_usd: number;
  budget_rm_display: number;
  spent_usd: number;
  spent_rm_display: number;
  remaining_usd: number;
  percent_used: number;
  warning_threshold_percent: number;
  warning: boolean;
  month_start: string;
  by_admin: Array<{ admin_id: number | null; username: string; estimated_cost_usd: number }>;
  daily: Array<{ date: string; estimated_cost_usd: number }>;
}

export async function getAIUsageSummary() {
  return (await api.get<AIUsageSummary>("/ai-usage/summary")).data;
}

export async function getAIUsage() {
  return (await api.get<AIUsageRecord[]>("/ai-usage")).data;
}
